app/apikey.go

```diff
@@ -0,0 +1,231 @@
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return NewAPIKeyClient(ct, conn), nil
+}
+
+func (s *APIKeyClient) createAPIKey(
+	displayName string,
+	description string,
+	expiry time.Time,
+	requestID string,
+) error {
+	expiryts, err := types.TimestampProto(expiry)
+	if err != nil {
+		return fmt.Errorf("failed to convert timestamp to proto: %s", err)
+	}
+	resp, err := s.authClient.CreateAPIKey(s.ctx, &authservice.CreateAPIKeyRequest{
+		Spec: &auth.APIKeySpec{
+			DisplayName: displayName,
+			Description: description,
+			ExpiryTime:  expiryts,
+		},
+		RequestId: requestID,
+	})
+	if err != nil {
+		return err
+	}
+	return PrintProto(resp)
+}
+
+func (s *APIKeyClient) listAPIKey() error {
+
+	totalRes := &authservice.GetAPIKeysResponse{}
+	pageToken := ""
+	for {
+		resp, err := s.authClient.GetAPIKeys(s.ctx, &authservice.GetAPIKeysRequest{
+			PageToken: pageToken,
+		})
+		if err != nil {
+			return err
+		}
+		totalRes.ApiKeys = append(totalRes.ApiKeys, resp.ApiKeys...)
+		// Check if we should continue paging
+		pageToken = resp.NextPageToken
+		if len(pageToken) == 0 {
+			return PrintProto(totalRes)
+		}
+	}
+}
+
+func (s *APIKeyClient) getAPIKey(
+	id string,
+) (*auth.APIKey, error) {
+	resp, err := s.authClient.GetAPIKey(s.ctx, &authservice.GetAPIKeyRequest{
+		Id: id,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.ApiKey, nil
+}
+
+func (s *APIKeyClient) deleteAPIKey(
+	id string,
+	resourceVersion string,
+	requestID string,
+) error {
+	resp, err := s.authClient.DeleteAPIKey(s.ctx, &authservice.DeleteAPIKeyRequest{
+		Id:              id,
+		ResourceVersion: resourceVersion,
+		RequestId:       requestID,
+	})
+	if err != nil {
+		return err
+	}
+	return PrintProto(resp)
+}
+
+func NewAPIKeyCommand(getAPIKeyClientFn GetAPIKeyClientFn) (CommandOut, error) {
+	var c *APIKeyClient
+	return CommandOut{
+		Command: &cli.Command{
+			Name:    "apikey",
+			Aliases: []string{"a"},
+			Usage:   "APIKey operations",
+			Before: func(ctx *cli.Context) error {
+				var err error
+				c, err = getAPIKeyClientFn(ctx)
+				return err
+			},
+			Subcommands: []*cli.Command{
+				{
+					Name:    "create",
+					Usage:   "Create an user apikey",
+					Aliases: []string{"c"},
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:     "name",
+							Usage:    "the display name of the apikey",
+							Required: true,
+							Aliases:  []string{"n"},
+						},
+						&cli.StringFlag{
+							Name:    "description",
+							Usage:   "the description of the apikey",
+							Aliases: []string{"desc"},
+						},
+						&cli.DurationFlag{
+							Name:    "duration",
+							Usage:   "the duration from now when the apikey will expire, will be ignored if expiry flag is set",
+							Aliases: []string{"d"},
+						},
+						&cli.TimestampFlag{
+							Name:    "expiry",
+							Usage:   "the absolute timestamp when the apikey will expire, example: 2023-04-12T15:14:05",
```

Comment:
[Nit] is this considered UTC time by default?

Reply:
I think the user needs to explicitly specify the time zone in the input, we should be good.